### PR TITLE
Remove .Result properties from `HackAndSlash` (and its likes)

### PR DESCRIPTION
**Pull request by longfin**
This PR removes `.Result` property to lighten the action, reducing the memory load.

### CODE REVIEW

**Comment by longfin** [comment]
I postponed removing `RankingBattle.Result` because it had been coupled with other states too tightly. 😓

**Review by ipdae — COMMENTED** [review]
it would be better to check simulation result using random seed.

**Comment by longfin** [comment]
> it would be better to check simulation result using random seed.

IMHO, if we want to check `StageSimulator.Result` on unit tests, `StageSimulatorTest` is a more proper place than `HackAndSlashTest` because `HackAndSlash` uses only `StageSimulator` and doesn't expose its result anymore.

File: Lib9c/Renderer/ActionRenderer.cs [hunk 1]
```diff
@@ -42,6 +42,7 @@ IAccountStateDelta nextStates
                 BlockIndex = context.BlockIndex,
                 OutputStates = nextStates,
                 PreviousStates = context.PreviousStates,
+                RandomSeed = context.Random.Seed
```
Comment (hskim881028) [inline Lib9c/Renderer/ActionRenderer.cs:45]: I get an error in 9c.
`'IRandom' does not contain a definition for 'Seed' and no accessible extension method 'Seed' accepting a first argument of type 'IRandom' could be found (are you missing a using directive or an assembly reference?)`

Comment (longfin) [reply Lib9c/Renderer/ActionRenderer.cs:45]: We need to update and merge [9c side](https://github.com/planetarium/NineChronicles/pull/627/) too after merging this PR.

**Review by hskim881028 — APPROVED** [review]

**Pull request merged**